test/README.md

```diff
@@ -1,95 +1,284 @@
+~~~ sh
+ubuntu_release=`lsb_release -rs`
+wget https://packages.microsoft.com/config/ubuntu/${ubuntu_release}/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+sudo dpkg -i packages-microsoft-prod.deb
+
+sudo apt-get install apt-transport-https
+sudo apt-get update
+sudo apt-get install msopenjdk-11
+~~~
+
+The steps above are for ubuntu 18.04. See [Install the Microsoft Build of the OpenJDK](https://docs.microsoft.com/en-us/java/openjdk/install) for information on how to install on other platforms.
+
+##### Setup access to Microsoft installation packages
+In particular, make sure to prepare your machine to access the Microsoft installation packages. *The steps to do this are listed below for your convenience. Again, please look at the section in the linked document for your os/arch combination if it is different from ubuntu18.04/amd64*
+
+_TODO: Access to Microsoft installation packages is required as this codebase depends on aziot-identity-service. Resolve this dependency by building from the identity service's code repo_
+
+~~~sh
+# Setup the repository information
+curl https://packages.microsoft.com/config/ubuntu/18.04/multiarch/prod.list > ./microsoft-prod.list
+
+# Setup the Microsoft GPG pulic key in the apt's trusted list.
+sudo cp ./microsoft-prod.list /etc/apt/sources.list.d/
+curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
+sudo cp ./microsoft.gpg /etc/apt/trusted.gpg.d/
+
+# Run update to update the package lists on your machine
+sudo apt-get update
+~~~
+
+##### Create Root CA Certificate for your machine
+Create a rootCA certificate for your machine *(See [Create Certs](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-create-test-certificates?view=iotedge-2020-11#create-root-ca-certificate) for more details)*. Navigate to the directory where the certificate generation scripts are found (typically iotedge/tools/CACertificates) and run
+~~~ sh
+./certGen.sh create_root_and_intermediate
+# This will create root and intermediate CA certs. The Root CA cert will be located in the 
+# certs sub directory and will be needed to run the tests.
+~~~
+
+##### Configure your local container repository
+We will be building the container images from the codebase and pushing them to a local container repository. To prepare, we will install the moby engine and configure the container repository using the steps below. 
+
+###### Install moby engine
+See [Install a container engine](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge?view=iotedge-2020-11#install-a-container-engine) for details
+
+~~~ sh
+sudo apt-get install moby-engine
+~~~
+###### Create TLS certificates
+We will be running the container registry with TLS and authentication turned on. If you already have certs for TLS, feel free to use those and skip the cert creation steps listed below. 

```

Comment:
Can any of the test certs generated in the earlier step be used here? If not, can our certGen.sh script be used to generate the certs in this step? If so, what args should be passed? If not, is this functionality something we want to add to certGen.sh in the future (i.e. should it be on our backlog)?

Reply:
IMO, these are unrelated. The certGen scripts are to create certs for device identity and communication, while these are TLS for the container registry. We should keep them separated out.